core/src/main/java/org/apache/carbondata/core/scan/result/iterator/AbstractDetailQueryResultIterator.java

```diff
@@ -94,6 +94,9 @@
     if (null != batchSizeString) {
       try {
         batchSize = Integer.parseInt(batchSizeString);
+        if (0 >= batchSize) {
```

Comment:
a) please make it more readable by using " batchSize <= 0"
b) Also I feel we should not validate here.  In CarbonReaderBuilder.withBatch(0), here validate the batch size and throw exception if it is less than or equal to zero

Reply:
Done!

Reply:
I have added the validation at CarbonProperties such that now both the flow of SDK and CarbonSession are covered in this case.